test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RegistryKeyUtilityTests.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Win32;
+using NuGet.Test.Utility;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace NuGet.CommandLine.Test
+{
+    public class RegistryKeyUtilityTests
+    {
+        private readonly TestLogger _logger;
+
+        public RegistryKeyUtilityTests(ITestOutputHelper output)
+        {
+            _logger = new TestLogger(output);
+        }
+
+        [WindowsNTFact]
+        public void GetValueFromRegistryKey_UnknownKey_ReturnsNull()
+        {
+            // Prepare
+            var guidString = new Guid().ToString();
+            var registryKey = RegistryKey.OpenBaseKey(RegistryHive.CurrentUser, RegistryView.Default);
+
+            // Act
+            var value = RegistryKeyUtility.GetValueFromRegistryKey(guidString, "key_local_machine\\test\\folder", registryKey, _logger);
+
+            // Assert
+            Assert.Null(value);
+            Assert.Empty(_logger.Messages);
+        }
+
+        [WindowsNTFact]
+        public void GetValueFromRegistryKey_GetSampleKey_ReturnsNotNull()
+        {
+            // Act
+            var value = RegistryKeyUtility.GetValueFromRegistryKey("ProductName", @"SOFTWARE\Microsoft\Windows NT\CurrentVersion", Registry.LocalMachine, _logger);
+
+            // Assert
+            Assert.NotNull(value);
+            var productName = value as string;
+            Assert.StartsWith("Windows", productName);
```

Comment:
Can Assert.StartsWith handle a null value? The implicit cast on the previous line will result in a null value if productName is anything other than a string. Can you double check that StartsWith can handle this otherwise we should have code that validates the productName is a string and do an explicit cast instead.

Reply:
`Assert.StartsWith` cannot handle null values. See https://github.com/xunit/assert.xunit/blob/37b46d188a5a8410cb2c15a208437b7f009a8068/StringAsserts.cs#L111

Added additional type assert and modified tests.

Reply:
The code you linked to suggests the otherwise. It doesn't matter if the value is null or not, if the strings don't match (or if either is null), the assert failure message will be the same.

Reply:
Oh, I misunderstood @jebriede question. Assert.StartsWith throw the same exception regardless or null/non-null string values.

Reply:
Thanks for checking!